tensorboard/components/tf_trace_viewer/tf-trace-viewer.html

```diff
@@ -15,6 +15,75 @@
 limitations under the License.
 -->
 
+<!-- Before any Trace Viewer scripts run, show warning if needed. -->
+<style>
+  .no-webcomponents-container.hidden {
+    display: none;
+  }
+
+  .no-webcomponents-container {
+    display: flex;
+    justify-content: center;
+    margin-top: 80px;
+    font-family: Roboto, sans-serif;
+  }
+
+  .no-webcomponents-container .title {
+    font-size: 22px;
+    font-weight: bold;
+  }
+
+  .no-webcomponents-container .contents {
+    max-width: 600px;
+  }
+</style>
+
+<div class="no-webcomponents-container hidden">
+  <div class="contents">
+    <div class="title">The trace viewer is currently unavailable.</div>
+    <p>
+      For more details, see
+      <a
+        href="https://github.com/tensorflow/tensorboard/issues/3209"
+        rel="noopener"
+        target="_blank"
+        >this issue</a
+      >.
+    </p>
+    <br />
+    <br />
+  </div>
+</div>
+
+<script>
+  const canUseWebComponentsV0 =
+    typeof document.registerElement === 'function' &&
+    typeof Element.prototype.createShadowRoot === 'function';
+  if (!canUseWebComponentsV0) {
+    const containerEl = document.querySelector('.no-webcomponents-container');
+    const contentsEl = document.querySelector(
+      '.no-webcomponents-container .contents'
+    );
+    containerEl.classList.remove('hidden');
+
+    const currentURL = new URL(window.location.href);
+    const relativeTraceURL = currentURL.searchParams.get('trace_data_url');
+    const traceURL = new URL(relativeTraceURL, window.location.href).toString();
+
+    const workaroundEl = document.createElement('p');
+    workaroundEl.textContent =
+      `If you are running a Chromium-based browser, ` +
```

Comment:
feels like this can be part of the DOM in L52. Can we just do that?

Reply:
Done.